.github/actions/python-setup/action.yml

```diff
@@ -0,0 +1,21 @@
+name: 'Python Setup'
+description: 'Install uv, python and dependencies'
+
+runs:
+  using: composite
+  steps:
+    - name: Install python
+      uses: actions/setup-python@v5
+
+    - name: Install uv
```

Comment:
While UV is still zero-version and making frequent breaking changes, I think it would be good to add a constraint on the UV version here.
I think everything else where we are using UV is using `>=0.5.12,<0.6.0`. To be reviewed post-May when we get back to tech debt and upgrades..

Reply:
..or possibly do it like
https://github.com/DemocracyClub/dc-data-baker/pull/23#discussion_r2024306809
if astral-sh/setup-uv respects that property?

Reply:
I've done this using the `required-version` section of `tool.uv` in pyproject.toml. I prefer this because it actually enforces it. I've stuck some of the limitations in a comment on the data baker pr.

Reply:
[150a563](https://github.com/DemocracyClub/dc-aws-org-tools/pull/1/commits/150a56378f79b059031c41129662db529a0c2cc8)

Reply:
Yeah I came back and added it after I started reading your other PR :)